pkgs/tools/package-management/libdnf/default.nix

```diff
@@ -0,0 +1,62 @@
+{ gcc9Stdenv, stdenv, fetchFromGitHub, cmake, gettext, pkgconfig, gpgme, libsolv, openssl, check, pcre, json_c, libmodulemd, libsmartcols, sqlite, librepo, libyaml, rpm }:
+
+gcc9Stdenv.mkDerivation rec {
+  pname = "libdnf";
+  version = "0.55.2";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = pname;
+    rev = version;
+    sha256 = "0hiydwfa90nsrqk5ffa6ks1g73wnsgjgq7z7gwq9jj76a7gpfbfq";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gettext
+    pkgconfig
+    check
+  ];
+
+  propagatedBuildInputs = [
+    gpgme
+    openssl
+    json_c
+    sqlite
+    librepo
+    pcre
+    rpm
+    libsolv
+    libsmartcols
+    libyaml
+    libmodulemd
+  ];
+
+  patches = stdenv.lib.optionals stdenv.isDarwin [ ./darwin.patch ];
+
+  prePatch = ''
+    # According to #cmake on freenode, libdnf should bundle the FindLibSolv.cmake module
```

Comment:
This does not make sense to me. What problem does this solve?

Reply:
I got this from https://github.com/NixOS/nixpkgs/blob/8d5eb691015063863e5f962268d62c24b73c6ca7/pkgs/tools/misc/rpm-ostree/default.nix#L105

Without this:

```
-- Usable gpgme flavours found:  vanilla pthread
CMake Error at CMakeLists.txt:53 (find_package):
  By not providing "FindLibSolv.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "LibSolv", but
  CMake did not find one.

  Could not find a package configuration file provided by "LibSolv"
  (requested version 0.6.30) with any of the following names:

    LibSolvConfig.cmake
    libsolv-config.cmake

  Add the installation prefix of "LibSolv" to CMAKE_PREFIX_PATH or set
  "LibSolv_DIR" to a directory containing one of the above files.  If
  "LibSolv" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "/tmp/nix-build-libdnf-0.55.2.drv-0/source/build/CMakeFiles/CMakeOutput.log".
builder for '/nix/store/zd5v6zmlm95iq7k3xa05rdyz2bv3hv3a-libdnf-0.55.2.drv' failed with exit code 1
error: build of '/nix/store/zd5v6zmlm95iq7k3xa05rdyz2bv3hv3a-libdnf-0.55.2.drv' failed
```

Reply:
Also https://github.com/rpm-software-management/libdnf/issues/113

Reply:
Could you add the link to the comment?

I verified that `${libsolv}` is in `CMAKE_PREFIX_PATH` and tried to move the `FindLibSolv.cmake` inside `libsolv` package to match various paths listed in https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure but still could not make CMake find it, not even in the simplified example:

```nix
with import <nixpkgs> { };
with pkgs;

let
  testSrc = writeText "CMakeLists.txt" ''
    project(libdnf C CXX)
    find_package(LibSolv 0.6.30 REQUIRED COMPONENTS ext)
  '';
in runCommandCC "cmake-find-test" {
    buildInputs = [ cmake libsolv ];
  } ''
    cp ${testSrc} CMakeLists.txt
    cmake .

    touch $out
  ''
```

Reply:
To which comment?
This thread or https://github.com/rpm-software-management/libdnf/issues/113 ?

Reply:
To the comment in the Nix file. Also I opened https://github.com/NixOS/nixpkgs/issues/107430, which might be better link.

Reply:
Added